lib/matplotlib/category.py

```diff
@@ -183,6 +186,21 @@ def update(self, data):
         """
         data = np.atleast_1d(np.array(data, dtype=object))
 
+        convertable = True
+        for val in data:
+            try:
+                float(val)
+            except ValueError:
+                try:
+                    dateutil.parser.parse(val)
+                except ValueError:
+                    convertable = False
+        if convertable:
+            cbook._warn_external('using category units to plot a list of strings '
```

Comment:
Can this be factored into an `__is_convertable` function? Both 'cause it'll be cleaner in the list below and easier to add more types to this if needed.

Reply:
done